### PR TITLE
[G2M] pagination - change item array to item length

### PR DESCRIPTION
**Pull request by kc-leung**
- please see storybook example for new usage in getting data and displaying them workflow

### CODE REVIEW

File: src/components/pagination.js [hunk 1]
```diff
@@ -35,19 +35,20 @@ const Pagination = ({ classes, items, onChangePage, onChangeRowsPerPage, initial
...
 
     if (page < 1 || page > _pager.totalPages) {
       return
     }
 
-    _pager = getPagerObject(items.length ? items.length : 0, page, rowsPerPageSize)
-    let pageOfItems = items.slice(_pager.startIndex, _pager.endIndex + 1)
+    _pager = getPagerObject(itemsLength ? itemsLength : 0, page, rowsPerPageSize)
+    const pageOfItems = []
+    for (let i = _pager.startIndex; i <= _pager.endIndex; i ++) pageOfItems.push({ i: i })
```
Comment (geoerika) [inline src/components/pagination.js:47]: Let's add `{}` for this `for` block, for readability.

File: stories/pagination.stories.js [hunk 2]
```diff
@@ -14,25 +14,26 @@ export const Normal = () => {
...
 
   return (
     <Layout>
       <Layout.Header className={headerClass}>Normal type with default styling</Layout.Header>
       <Layout.Content className={contentClass}>
-        {data.pageOfItems.map(item =>
-          <div key={item.id}>{item.name}</div>,
-        )}
+        {data.pageOfItems.map((item, index) => {
+          return (<div key={index}>{data.exampleData[item.i].name}</div>)
```
Comment (geoerika) [inline stories/pagination.stories.js:32]: No need to add return here. Can simplify:
`
{data.pageOfItems.map((item, index) => (
  <div key={index}>{data.exampleData[item.i].name}</div>
))}
`

File: stories/pagination.stories.js [hunk 3]
```diff
@@ -43,25 +44,26 @@ export const NoAttachments = () => {
...
 
   return (
     <Layout>
       <Layout.Header className={headerClass}>with no pages, first & last, rows per page attachments</Layout.Header>
       <Layout.Content className={contentClass}>
-        {data.pageOfItems.map(item =>
-          <div key={item.id}>{item.name}</div>,
-        )}
+        {data.pageOfItems.map((item, index) => {
+          return (<div key={index}>{data.exampleData[item.i].name}</div>)
```
Comment (geoerika) [inline stories/pagination.stories.js:62]: See above.

**Review by geoerika — APPROVED** [review]
Minor comments!

**Review by hyx131 — APPROVED** [review]

**Review by DoParkEQ — APPROVED** [review]

**Pull request merged**